internal/index/manager/manager.go

```diff
@@ -784,41 +785,66 @@ func (mgr *Manager) UpdateTag(name string, operation UpdateTagOperation) error {
 				if maxUsedStreamID >= mgr.nextStreamID {
 					return fmt.Errorf("unknown stream id %d", maxUsedStreamID)
 				}
-				newTag := *t
```

Comment:
why did you remove the copy of the tag?
this is essentially implementing copy-on-write, meaning other places can safely keep a reference to the tags without having to deal with these tags changing...

Reply:
(i doubt that this part here is what was expensive)

Reply:
(this part = the copying of the tag, not the stuff below, which likely was very expensive)